tests/test_decorators.py

```diff
@@ -518,3 +518,33 @@ def validate_many(self, data, many):
         errors = schema.validate([{'foo': 3, 'bar': 'not an int'}], many=True)
         assert 'bar' in errors[0]
         assert '_schema' not in errors
+
+    def test_allow_reporting_field_errors_in_schema_validator(self):
+
+        class NestedSchema(Schema):
+            baz = fields.Int(required=True)
+
+        class MySchema(Schema):
+            foo = fields.Int(required=True)
+            bar = fields.Nested(NestedSchema, required=True)
+            bam = fields.Int(required=True)
+
+            @validates_schema(skip_on_field_errors=True)
+            def consistency_validation(self, data):
+                errors = {}
+                if data['bar']['baz'] != data['foo']:
+                    errors['bar'] = {'baz': 'Non-matching value'}
+
+                if data['bam'] > data['foo']:
+                    errors['bam'] = 'Value should be less than foo'
+
+                if errors:
+                    raise ValidationError(errors)
+
+        schema = MySchema()
+        errors = schema.validate({'foo': 2, 'bar': {'baz': 5}, 'bam': 6})
```

Comment:
Before this PR `errors` would have been

```py
{'_schema': [{'bam': 'Value should be less than foo', 'bar': {'baz': 'Non-matching value'}}]}
```

but now is

```py
{'bam': 'Value should be less than foo', 'bar': {'baz': 'Non-matching value'}}
```

The `_schema` key is documented in the [Schema Level Validation](https://marshmallow.readthedocs.io/en/dev/extending.html#schema-level-validation) section of the Extending page.

Assigning schema validation errors to specific fields is documented in the [Storing Errors on Specific Fields](https://marshmallow.readthedocs.io/en/dev/extending.html#storing-errors-on-specific-fields) section.